### PR TITLE
mkcloud: stop using loop devices for partition access

### PR DESCRIPTION
**Pull request by bmwiedemann**
this means we no more need to use the offset of partitions

code worked for me in simple tests

### CODE REVIEW

**Comment by MaximilianMeister** [comment]
+1 codewise from my side (haven't tested it though)

**Comment by aspiers** [comment]
At a glance, this looks very nice :)

**Comment by rsalevsky** [comment]
+1 tested basically with an plain mkcloud run

File: scripts/mkcloud [hunk 1]
```diff
@@ -396,13 +384,17 @@ function onhost_deploy_image()
...
-        local loopdevice=$(losetup --find --show -o $(( $offset * 512 )) $disk)
-        safely fsck -y -f $loopdevice
-        safely resize2fs $loopdevice
-        losetup -d $loopdevice
+        part2=$(kpartx -asv $disk|perl -ne 'm/add map (\S+2) / && print $1')
+        test -n "$part2" || complain 31 "failed to find partition #2"
+        local bdev=/dev/mapper/$part2
+        safely fsck -y -f $bdev
+        safely resize2fs $bdev
+        sleep 1 # time for dev to become unused
```
Comment (dirkmueller) [inline scripts/mkcloud]: blockdev --flushbufs $bdev

Comment (bmwiedemann) [inline scripts/mkcloud]: that does not help. it still errors with
device-mapper: remove ioctl on mkcvg0-g15.admin2 failed: Device or resource busy

**Pull request merged**